neat_ml/utils/figure_utils.py

```diff
@@ -0,0 +1,376 @@
+    - both axes MajorLocator(step)
+    - both axes FormatStrFormatter('%.0f')
+    """
+    ax.set_xlim(xrange)
+    ax.set_ylim(yrange)
+    ax.set_aspect("auto")
+
+    for spine in ax.spines.values():
+        spine.set_linewidth(2)
+    ax.tick_params(
+        direction="in",
+        length=5,
+        width=2,
+        which="both",
+        top=True,
+        right=True,
+        labelsize=24,
+    )
+
+def plot_gmm_decision_regions(
+    df: pd.DataFrame,
+    x_col: str,
+    y_col: str,
+    phase_col: str,
+    ax,
+    xrange: Sequence[float],
+    yrange: Sequence[float],
+    n_components: int,
+    random_state: int,
+    boundary_color: str,
+    resolution: int,
+    decision_alpha: float,
+    plot_regions: bool,
+    region_colors: Optional[list[str]],
+) -> tuple[GaussianMixture, np.ndarray, Optional[np.ndarray]]:
+    """
+    Trains a GMM and creates contour traces for phase 
+    regions and boundaries.
+
+    Parameters:
+    -----------
+        df (pd.DataFrame): DataFrame with composition and
+                           phase data.
+        x_col (str): Name of the column for the x-axis 
+                     component.
+        y_col (str): Name of the column for the y-axis 
+                     component.
+        phase_col (str): Name of the column containing 
+                         phase information.
+        xrange (list[int]): X-axis composition range.
+        yrange (list[int]): Y-axis composition range. 
+        n_components (int): Number of GMM components.
+        random_state (int): Seed for reproducibility.
+        region_colors (Optional[list[str]]): Colors for the 
+                                             phase regions.
+        boundary_color (str): Color for the boundary line.
+        resolution (int): Grid resolution for the contour plot.
+        decision_alpha (float): Opacity of the filled regions.
+        plot_regions (bool): Whether to plot the filled regions.
+
+    Returns:
+    --------
+        tuple: Containing the GMM model, cluster labels, 
+               boundary points, and a list of Plotly contour
+               traces.
+    """
+    df_local = df.copy()
+    x_features = df_local[phase_col].to_numpy().reshape(-1, 1)
+
+    gmm = GaussianMixture(n_components=n_components, random_state=random_state)
+    raw_labels = gmm.fit_predict(x_features)
+
+    x_comp = df_local[[x_col, y_col]].to_numpy()
+    std_labels, label_map = _standardise_labels(raw_labels, x_comp)
+    wrapper = GMMWrapper(gmm, x_comp, x_features)
+    x_start, x_stop = map(float, xrange)
+    y_start, y_stop = map(float, yrange)
+    xs = np.linspace(x_start, x_stop, resolution)
+    ys = np.linspace(y_start, y_stop, resolution)
+    xx, yy = np.meshgrid(xs, ys)
+    z_raw = wrapper.predict(np.c_[xx.ravel(), yy.ravel()])
+    z = np.vectorize(label_map.get)(z_raw).reshape(xx.shape)
+
+    if region_colors is None:
+        region_colors = ["aquamarine", "lightsteelblue"]
```

Comment:
Not hit by regression tests:

<img width="538" height="56" alt="Image" src="https://github.com/user-attachments/assets/37018bef-0c54-4bab-bd33-af69eb5c78cf" />